src/v/datalake/coordinator/iceberg_snapshot_remover.cc

```diff
@@ -56,11 +57,42 @@ snapshot_remover::errc log_and_convert_action_errc(
 ss::future<checked<std::nullopt_t, snapshot_remover::errc>>
 iceberg_snapshot_remover::remove_expired_snapshots(
   model::topic topic, model::timestamp ts) const {
-    auto table_id = table_id_for_topic(topic);
+    // Kick off removals from both the main and DLQ tables.
+    auto main_fut = remove_expired_snapshots(
+      topic, table_id_provider::table_id(topic), ts);
+    auto dlq_fut = remove_expired_snapshots(
+      topic, table_id_provider::dlq_table_id(topic), ts);
+
+    std::optional<snapshot_remover::errc> error;
+    auto dlq_res = co_await std::move(dlq_fut);
```

Comment:
is there any chance that both of these futures could be exceptional? if so, then co_await dlq_fut would throw, and main_fut would be an abandoned exceptional future. or if the first were to throw, then the second would effectively become future running in the background.

Reply:
In theory we shouldn't because (I've tried to maintain that) we don't throw from most/all methods that return a result/checked type. But in any case, I'll update this to just do these in series to avoid the concern, since doing it in parallel is somewhat a premature optimization.